lib/premailer/adapter/rgb_to_hex.rb

```diff
@@ -0,0 +1,31 @@
+# RGB helper for adapters, currently only nokogiri supported
+
+module AdapterHelper
+  module RgbToHex
+    def to_hex(str)
+      str.to_i.to_s(16).rjust(2, '0').upcase
+    end
+
+    def is_rgb?(color)
+      pattern = %r{
+      rgb
+      \(          # literal open
+      (\d{1,3})   # capture 1-3 digits
+      \s*,\s*     # comma, with optional whitespace
+      (\d{1,3})   # capture 1-3 digits
+      \s*,\s*     # comma, with optional whitespace
+      (\d{1,3})   # capture 1-3 digits
+      \)          # literal close
+      }x
```

Comment:
Potentially whitespace inside the parentheses too, it might be worth covering that although not likely to occur.

Another possibility is `rgba(4,3,2,1)` for transparent colors (alpha channel) but let's just stick to plain RGB for now.